### PR TITLE
[SQL Migration] TDE experience addition for New Migration Experience

### PR DESCRIPTION
**Pull request by stuti149**
This PR:

- adds experience when an encrypted database is selected.
- corrects icon for ready database
- adds bottom border for body to improve UI experience.

Screenshots:

1. When user selects MI as target and an encrypted database is selected, dialog opens up.
image

2. When one or more encrypted databases are present, ability to add certificates is given in instance summary.
image

3.  When user selects encrypted database. Encryption column is added in summary for that particular database.
image

### CODE REVIEW

File: extensions/sql-migration/src/wizard/assessmentDetailsPage/assessmentDetialsBody.ts [hunk 1]
```diff
@@ -19,33 +19,33 @@ import { SqlMigrationAssessmentResultItem } from '../../service/contracts';
...
+		this._instanceSummary = new InstanceSummary(migrationStateModel);
 	}
 
 	public get treeComponent() {
 		return this._treeComponent;
 	}
 
 	// function that defines all the components for body section
-	public createAssessmentDetailsBody(view: azdata.ModelView): azdata.Component {
+	public async createAssessmentDetailsBody(view: azdata.ModelView): Promise<azdata.Component> {
```
Comment (Ramudaykumar) [inline extensions/sql-migration/src/wizard/assessmentDetailsPage/assessmentDetialsBody.ts:41]: Async should be added to the end of the async method name.

File: extensions/sql-migration/src/wizard/assessmentDetailsPage/assessmentDetialsBody.ts [hunk 2]
```diff
@@ -106,7 +105,7 @@ export class AssessmentDetailsBody {
 	}
 
 	// creates ui for summary component based on selections in left panel.
-	private createSummaryComponent(): azdata.FlexContainer {
+	private async createSummaryComponent(): Promise<azdata.FlexContainer> {
```
Comment (Ramudaykumar) [inline extensions/sql-migration/src/wizard/assessmentDetailsPage/assessmentDetialsBody.ts:108]: Same here for Async method.

File: extensions/sql-migration/src/wizard/assessmentDetailsPage/instanceSummary.ts [hunk 3]
```diff
@@ -23,29 +28,39 @@ export class InstanceSummary {
...
+	private _tdeConfigurationDialog!: TdeConfigurationDialog;
+	private _previousMiTdeMigrationConfig: TdeMigrationModel = new TdeMigrationModel(); // avoid null checks
+	private _tdeDatabaseSelectedHelperText!: azdata.TextComponent;
+	private _tdeEditButton!: azdata.ButtonComponent;
+
+	constructor(public migrationStateModel: MigrationStateModel) { }
 
 	// function to create instance summary components
-	public createInstanceSummaryContainer(view: azdata.ModelView): azdata.FlexContainer {
+	public async createInstanceSummaryContainer(view: azdata.ModelView): Promise<azdata.FlexContainer> {
```
Comment (Ramudaykumar) [inline extensions/sql-migration/src/wizard/assessmentDetailsPage/instanceSummary.ts:41]: Same here.

File: extensions/sql-migration/src/wizard/assessmentDetailsPage/instanceSummary.ts [hunk 4]
```diff
@@ -63,25 +77,29 @@ export class InstanceSummary {
...
+	public async populateInstanceSummaryContainer(): Promise<void> {
+		this._previousMiTdeMigrationConfig = this.migrationStateModel.tdeMigrationConfig;
+		await this.refreshTdeView();
+		this._assessedDatabases.value = constants.ASSESSED_DBS_LABEL + ": " + this.migrationStateModel._databasesForAssessment?.length;
+		this._totalFindingLabels.value = constants.TOTAL_FINDINGS_LABEL + ": " + this.migrationStateModel._assessmentResults?.issues.filter(issue => issue.appliesToMigrationTargetPlatform === this.migrationStateModel._targetType).length;
+		const readyDbsCount = this.migrationStateModel._assessmentResults.databaseAssessments.filter((db) => db.issues.filter(issue => issue.appliesToMigrationTargetPlatform === this.migrationStateModel._targetType).length === 0).length;
+		const notReadyDbsCount = this.migrationStateModel._assessmentResults.databaseAssessments.filter((db) => db.issues.filter(issue => issue.appliesToMigrationTargetPlatform === this.migrationStateModel._targetType && issue.issueCategory === IssueCategory.Issue).length !== 0).length;
+		const readyWithWarnDbsCount = this.migrationStateModel._databasesForAssessment?.length - (readyDbsCount + notReadyDbsCount);
```
Comment (krritik) [inline extensions/sql-migration/src/wizard/assessmentDetailsPage/instanceSummary.ts:102]: this.migrationStateModel._assessmentResults.databaseAssessments is used multiple times. We can store this in a separate variable say totalDbs and use it again.

File: extensions/sql-migration/src/wizard/assessmentDetailsPage/assessmentDetialsBody.ts (hunk 1 above)
Comment (stuti149) [reply extensions/sql-migration/src/wizard/assessmentDetailsPage/assessmentDetialsBody.ts]: Changed method name

File: extensions/sql-migration/src/wizard/assessmentDetailsPage/instanceSummary.ts (hunk 4 above)
Comment (stuti149) [reply extensions/sql-migration/src/wizard/assessmentDetailsPage/instanceSummary.ts:104]: done

File: extensions/sql-migration/src/wizard/assessmentDetailsPage/instanceSummary.ts (hunk 3 above)
Comment (stuti149) [reply extensions/sql-migration/src/wizard/assessmentDetailsPage/instanceSummary.ts]: done

File: extensions/sql-migration/src/wizard/assessmentDetailsPage/assessmentDetialsBody.ts (hunk 2 above)
Comment (stuti149) [reply extensions/sql-migration/src/wizard/assessmentDetailsPage/assessmentDetialsBody.ts]: done

**Review by Ramudaykumar — APPROVED** [review]

**Pull request merged**